### PR TITLE
Drop `server.R`

### PR DESCRIPTION
**Pull request by schloerke**
Should be merged into #2821 before #2821 is merged into master

* Produces a full `adhoc.R` file that uses `withr::with_environment()` + `loadSupport()` and has dynamic content.
* Removed `helper-support.R` as `testthat::test_dir` now calls `loadSupport` directly
* Removed regular shiny package testing output
* Fixed broken lexical sort test
* Added context to all `testthat` files

### CODE REVIEW

**Comment by wch** [comment]
I think we can drop adhoc.R -- we can document the capability but not include it in the template.

File: R/app_template.R [hunk 1]
```diff
@@ -38,25 +37,21 @@
...
 #' * `tests/testthat.R` is a test runner for test files in the
-#'   `tests/testthat/` directory.
-#' * `tests/testthat/helper-load.R` is a helper script that is automatically
-#'   loaded before running `test-mymodule.`R. (This is performed by the testthat
-#'   package.)
-#' * `tests/testthat/test-sort.R` is a set of tests that use the
-#'   [testthat](https://testthat.r-lib.org/) package for testing.
+#'   `tests/testthat/` directory using the [testthat](https://testthat.r-lib.org/) package.
+#' * `tests/testthat/test-mymodule.R` is a test for the module.
+#' * `tests/testthat/test-server.R` is a test for the appliction.
```
Comment (wch) [inline R/app_template.R:48]: For this line and above, say "is a test for the application's server code."

File: R/app_template.R [hunk 2]
```diff
@@ -38,25 +37,21 @@
...
+#' * `tests/testthat/test-sort.R` is a test for a supporting function.
```
Comment (wch) [inline R/app_template.R:49]: "... in the R/ directory."

File: R/app_template.R [hunk 3]
```diff
@@ -38,25 +37,21 @@
...
+#' * `tests/adhoc.R` is a DIY testing file that produces an error if a test fails.
```
Comment (wch) [inline R/app_template.R:50]: Remove this.

File: R/app_template.R [hunk 4]
```diff
@@ -142,7 +137,7 @@ shinyAppTemplate <- function(path = NULL, examples = "default")
 
     # Filter out files related to R/sort.R, if applicable.
     if (!with_rdir) {
-      files <- files[!grepl("utils", files)]
+      files <- files[!grepl("sort", files)]
```
Comment (wch) [inline R/app_template.R:140]: Replace with a search for files in R/ that are not modules.

File: inst/app_template/app.R [hunk 5]
```diff
@@ -35,6 +35,7 @@ if (isTRUE(module)) {
 }
 }}
   data <- reactive({
+    req(input$size)
```
Comment (wch) [inline inst/app_template/app.R:38]: Drop this - we should make the test runner behave more like a real app.

File: inst/app_template/tests/testthat/test-mymodule.R [hunk 6]
```diff
@@ -1,8 +1,7 @@
-# Use testthat just for expectations
-library(testthat)
+context("mymoduleServer")
 
 # See ?testServer for more information
-testServer(mymoduleServer, {
+testServer(mymoduleServer, expr = {
```
Comment (wch) [inline inst/app_template/tests/testthat/test-mymodule.R:4]: Remove `expr =`

File: inst/app_template/tests/testthat/test-server.R [hunk 7]
```diff
@@ -1,11 +1,11 @@
-# Use testthat just for expectations
-library(testthat)
+context("app")
 
-testServer('../..', {
+testServer('../..', expr = {
```
Comment (wch) [inline inst/app_template/tests/testthat/test-server.R:3]: Proposed syntax:

```
testServer(expr, server = '.', args = list())

# Testing app:
testServer({
   ...
})

#Testing module:
testServer(server = mymodule, args = list(x=1, y=2), {
    ....
})
```

File: inst/app_template/tests/testthat/test-server.R [hunk 8]
```diff
@@ -1,11 +1,11 @@
...
+  # TODO-barret remove path
   # Set the `size` slider and check the output
   session$setInputs(size = 6)
   expect_equal(output$sequence, "1 2 3 4 5 6")
 
   session$setInputs(size = 12)
-  expect_equal(output$sequence, "1 2 3 4 5 6 7 8 9 10 11 12")
+  expect_equal(output$sequence, "1 10 11 12 2 3 4 5 6 7 8 9")
```
Comment (wch) [inline inst/app_template/tests/testthat/test-server.R:10]: This also needs to be conditional on whether sort.R is included.

File: tests/testthat/test-test.R [hunk 9]
```diff
@@ -141,3 +145,12 @@ test_that("runTests runs as expected without rewiring", {
   ))
   expect_s3_class(df, "shiny_runtests")
 })
+
+
+test_that("app template works with runTests", {
+  tempTemplateDir <- file.path(tempdir(), paste0("shinyAppTemplate-", floor(runif(1) * 10000)))
+  shinyAppTemplate(tempTemplateDir, "all")
```
Comment (wch) [inline tests/testthat/test-test.R:152]: Would be good to test with different settings as well, notably the different combinations of `rdir` and `module`.

**Comment by schloerke** [comment]
Closing in favor of #2827

**Comment by schloerke** [comment]
Deleting branch as newer branch was merged

**Pull request closed without merging**